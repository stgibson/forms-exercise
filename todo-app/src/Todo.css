.Todo {
  margin-bottom: 5px;
}

.Todo-button {
  margin-left: 5px;
}