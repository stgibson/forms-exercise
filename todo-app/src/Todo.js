import React from "react";
import "./Todo.css";

const Todo = ({ id, task, showEditForm, removeTodo }) => {
  return (
    <div id={ id } className="Todo">
      { task }
      <button className="Todo-button" onClick={ showEditForm }>edit</button>
      <button className="Todo-button" onClick={ removeTodo }>X</button>
    </div>
  );
};

export default Todo;