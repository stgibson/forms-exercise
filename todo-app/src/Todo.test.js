import React from "react";
import Todo from "./Todo";
import { render } from "@testing-library/react";

it("renders without crashing", () => {
  render(<Todo task="Test Task" />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Todo task="Test Task" />);
  expect(asFragment()).toMatchSnapshot();
});