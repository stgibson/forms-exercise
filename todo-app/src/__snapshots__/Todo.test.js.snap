// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    Test Task
    <button
      class="Todo-button"
    >
      edit
    </button>
    <button
      class="Todo-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
